waiter/src/waiter/core.clj

```diff
@@ -1765,11 +1783,13 @@
                               (wrap-secure-request-fn
                                 (fn token-owners-handler-fn [request]
                                   (token/handle-list-token-owners-request kv-store request))))
-   :token-refresh-handler-fn (pc/fnk [[:state kv-store]
+   :token-refresh-handler-fn (pc/fnk [[:daemons token-watch-maintainer]
+                                      [:state kv-store]
                                       wrap-router-auth-fn]
                                (wrap-router-auth-fn
                                  (fn token-refresh-handler-fn [request]
-                                   (token/handle-refresh-token-request kv-store request))))
+                                   (token/handle-refresh-token-request kv-store (:tokens-update-chan token-watch-maintainer)
```

Comment:
Nitpick, extract this in a let-block, {:keys [tokens-update-chan]} token-watch-maintainer.